src/core/issues-notes.js

```diff
@@ -280,17 +293,48 @@ function createLabel(label, repoURL) {
     href="${issuesURL.href}">${name}</a>`;
 }
 
+/**
+ * @param {string} githubAPI
+ * @returns {Promise<Map<string, GitHubIssue>>}
+ */
+async function fetchAndStoreGithubIssues(githubAPI) {
+  if (!githubAPI) {
+    return new Map();
+  }
+
+  /** @type {NodeListOf<HTMLElement>} */
+  const specIssues = document.querySelectorAll(".issue[data-number]");
+  const issueNumbers = [...specIssues]
+    .map(elem => Number.parseInt(elem.dataset.number, 10))
+    .filter(issueNumber => issueNumber);
+
+  if (!issueNumbers.length) {
+    return new Map();
+  }
+
+  const url = new URL("issues", githubAPI);
+  url.searchParams.set("issues", issueNumbers.join(","));
+
+  const response = await fetchAndCache(url.href);
```

Comment:
A regular fetch might be enough here, as server is sending cache headers already